core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -492,6 +454,8 @@ sealed abstract class Doc extends Product with Serializable {
     if (res._2) Some(res._1) else None
   }
 
+  def flatten = flattenOption getOrElse this
```

Comment:
Can we add a return type and move the comment down?

Reply:
Done.   I was not sure about keeping the comment, since that can't be true if we implement @olafurpg's request and keep the existing type of `flatten`, but it's definitely the right call for now.

I amended my commit and force pushed to my branch, which at some point in the past would update the pull request.  That seems to no longer be true, or I messed it up in some way, so I got a new PR: #98.   I will try to figure out what I did wrong in git in the meantime and close this request.